modules/exploits/windows/http/nscp_authenticated_rce.rb

```diff
@@ -0,0 +1,219 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include ::Msf::Exploit::Remote::HttpClient
+  include ::Msf::Exploit::CmdStager
+  include ::Msf::Exploit::Powershell
+  prepend ::Msf::Exploit::Remote::AutoCheck
+  include ::Rex::Text
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'NSClient++ 0.5.2.35 - Authenticated Remote Code Execution',
```

Comment:
I presume the functionality exploited in this module is an intended feature? It looks like you're uploading a plugin, which presumably can execute arbitrary code by design. If so, the version number isn't relevant for the module title.

I haven't looked into the feature / "vulnerability" at all, but perhaps `NSClient++ ExternalScripts Authenticated Remote Code Execution` is a better title, as `ExternalScripts` is a necessary (and non-default?) component for exploitation.

Reply:
Yes it is non-default, I added this to the name.